Library/Homebrew/software_spec.rb

```diff
@@ -443,13 +441,18 @@ def prefix=(prefix)
 
   def root_url(var = nil, specs = {})
     if var.nil?
-      @root_url ||= if Homebrew::EnvConfig.bottle_domain.start_with?(GitHubPackages::URL_PREFIX)
+      @root_url ||= if Homebrew::EnvConfig.bottle_domain.match?(GitHubPackages::URL_REGEX)
         GitHubPackages.root_url(tap.user, tap.repo).to_s
       else
         "#{Homebrew::EnvConfig.bottle_domain}/#{Utils::Bottles::Bintray.repository(tap)}"
       end
     else
-      @root_url = var
+      @root_url = if var.to_s.start_with? "docker://"
```

Comment:
The `root_url` should never start with `docker://`. When is it set to that?

Reply:
In Brewsci/bio I set `root_url "docker://ghcr.io/brewsci/bio"`. It also supports other Docker / OCI / ORAS container registries such as `docker://hub.docker.com` and `docker://quay.io`.

Reply:
@sjackman Ok. Can this logic get moved to `github_packages.rb`? Ideally this would just be part of the `root_url` method.

Reply:
I'm not entirely sure that I followed, but I took a stab at it in PR https://github.com/Homebrew/brew/pull/11037